go/oasis-node/cmd/common/metrics/resource.go

```diff
@@ -0,0 +1,61 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/spf13/viper"
+
+	"github.com/oasislabs/oasis-core/go/common/service"
+)
+
+// ResourceCollector is interface for monitoring resources (cpu, mem, disk, net).
+type ResourceCollector interface {
+	// Update updates corresponding resource metrics.
+	Update() error
+}
+
+// resourceService updates and collects number of resources.
```

Comment:
Mention the interval here instead as this is what actually implements periodic collection.